File: fe/mobile/android/app/src/main/java/com/capstone/bridge/NotificationBridgePackage.java
package com.capstone.bridge;

import com.capstone.domain.repository.NotificationRepository;
import com.capstone.infrastructure.security.PermissionManager;
import com.facebook.react.bridge.NativeModule;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.ReactPackage;
import com.facebook.react.uimanager.ViewManager;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class NotificationBridgePackage implements ReactPackage {
    private final NotificationRepository repository;
    private final PermissionManager permissionManager;

    public NotificationBridgePackage(NotificationRepository repository, PermissionManager permissionManager) {
        this.repository = repository;
        this.permissionManager = permissionManager;
    }

    @Override
    public List<NativeModule> createNativeModules(ReactApplicationContext reactContext) {
        List<NativeModule> modules = new ArrayList<>();
        modules.add(new NotificationBridgeModule(reactContext, repository, permissionManager));
        return modules;
    }

    @Override
    public List<ViewManager> createViewManagers(ReactApplicationContext reactContext) {
        return Collections.emptyList();
    }
}

File: fe/mobile/android/app/src/main/java/com/capstone/bridge/NotificationBridgePackage.kt
package com.capstone.bridge

import android.view.View
import com.capstone.domain.repository.NotificationRepository
import com.facebook.react.ReactPackage
import com.facebook.react.bridge.NativeModule
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.uimanager.ReactShadowNode
import com.facebook.react.uimanager.ViewManager

class NotificationBridgePackage(
    private val repository: NotificationRepository
) : ReactPackage {
    override fun createNativeModules(reactContext: ReactApplicationContext): List<NativeModule> {
        return listOf(NotificationBridgeModule(reactContext, repository))
    }

    override fun createViewManagers(reactContext: ReactApplicationContext): List<ViewManager<View, ReactShadowNode<*>>> {
        return emptyList()
    }
}
